components/WeatherForecastModal.js

```diff
@@ -0,0 +1,92 @@
+import {
+  WeatherForecastModalField,
+  ModalCloseButton,
+  Table,
+  DisplayedDaysContainer,
+  DisplayedDay,
+  Date,
+  Daytime,
+  WeatherDataContainer,
+  WeatherIcon,
+  Temperature,
+  Windspeed,
+} from "./styledComponents/WeatherForecasModal.styles";
+import WeekdayFromDateString from "./WeekdayFromDateString";
+import ArrowIcon from "./icons/ArrowWeatherIcon";
+
+export default function WeatherForecastModal({
+  filteredWeatherMorning,
+  filteredWeatherAfternoon,
+  filteredWeatherEvening,
+  onClickButton,
+}) {
+  // extracting the dates from the date-time-String with help of the static end value "06:00:00"
+  const dates = filteredWeatherMorning
+    .filter((date) => date.dt_txt.endsWith("06:00:00"))
+    .map((date) => date.dt_txt.split(" ")[0]);
+
+  return (
+    <>
+      <WeatherForecastModalField>
+        <ModalCloseButton onClick={onClickButton}>x</ModalCloseButton>
+        <Table>
+          <tbody>
+            <td>
+              {/*because of space issues and of the everchanging WeatherArray, the filteredWeatherArrays will only be mapped to the third day  */}
+              {filteredWeatherMorning.slice(0, 3).map((weekday, index) => (
+                <DisplayedDaysContainer key={index}>
+                  <DisplayedDay>
+                    <WeekdayFromDateString dateString={weekday.dt_txt} />
+                  </DisplayedDay>
+                  <Date>{dates[index]}</Date>
+                </DisplayedDaysContainer>
+              ))}
+            </td>
+            <td>
+              <Daytime>6am</Daytime>
+              {filteredWeatherMorning.slice(0, 3).map((weather, index) => (
+                <WeatherDataContainer key={index}>
+                  <WeatherIcon
+                    alt="Weatherdata for Morning"
+                    src={`weatherIcons/${weather.weather[0].icon}.png`}
+                  />
+                  <Temperature>{Math.round(weather.main.temp)}°C</Temperature>
+                  <ArrowIcon deg={weather.wind.deg} />
+                  <Windspeed>{weather.wind.speed} m/s</Windspeed>
+                </WeatherDataContainer>
+              ))}
+            </td>
+            <td>
+              <Daytime>12am</Daytime>
+              {filteredWeatherAfternoon.slice(0, 3).map((weather, index) => (
+                <WeatherDataContainer key={index}>
+                  <WeatherIcon
+                    alt="Weatherdata for Afternoon"
+                    src={`weatherIcons/${weather.weather[0].icon}.png`}
+                  />
+                  <Temperature>{Math.round(weather.main.temp)}°C</Temperature>
+                  <ArrowIcon deg={weather.wind.deg} />
+                  <Windspeed>{weather.wind.speed} m/s</Windspeed>
+                </WeatherDataContainer>
+              ))}
```

Comment:
you are doing the same thing three times here. Couldn't you create a component and reuse it? Alternatively, you could remove this modal component and display the table directly in the WeatherForecast component. Then you could create a generic DailyWeatherDisplay component which takes an array of weather data and a title so you can use the one component for morning afternoon and evening.

Reply:
With the implementation of the following change, this point has become less relevant.